controller.go

```diff
@@ -217,6 +217,26 @@ func (c *controller) initDiscovery(watcher discovery.Watcher) error {
 		return fmt.Errorf("discovery initialization requires a valid configuration")
 	}
 
+	addr := strings.Split(c.cfg.Cluster.Address, ":")
+	advIP := net.ParseIP(addr[0])
+	if advIP == nil {
+		return fmt.Errorf("invalid cluster advertise configuration (%s)", c.cfg.Cluster.Address)
+	}
+
+	addrs, err := net.InterfaceAddrs()
+	if err != nil {
+		log.Errorf("Unable to get interface addresses %v", err)
+	} else {
+		for _, addr := range addrs {
+			ip, _, err := net.ParseCIDR(addr.String())
+			if err == nil && ip.Equal(advIP) {
+				goto discover
+			}
+		}
+		log.Errorf("Multi-Host networking requires cluster-advertise(%s) to be configured with a local ip-address that is reachable within the cluster", advIP.String())
```

Comment:
Since we don't have a way change the kv-store config on the fly shouldn't return an error and let the daemon error out giving the user a chance to fix the problem right away ?